include/net/socket.h

```diff
@@ -688,8 +688,23 @@ static inline ssize_t recv(int sock, void *buf, size_t max_len, int flags)
 	return zsock_recv(sock, buf, max_len, flags);
 }
 
+/*
+ * Need this wrapper because newer GCC versions got too smart and "typecheck"
+ * even macros, so '#define fcntl zsock_fcntl' leads to error.
+ */
+static inline int _zsock_fcntl_wrapper(int sock, int cmd, ...)
```

Comment:
I am not sure, but just wondering if this cause some issue in misra checks as we tried to move away from using the `_` prefix. We could use `zsock_fcntl_wrapper()` directly here without any extra prefix just to be in safe side.

Reply:
> move away from using the _ prefix.

Fixed. Just tried to signify it's internal-only function.